### ISSUE TITLE
Message popup is not announced and read by screen readers

### ISSUE THREAD

**Issue by ali-kashanian** [paypal/paypal-messaging-components#927]
Hello,

Our team is using the messaging component on the shopping cart page and we have recently had our website undergone for ADA compliance audit. In one of the defects they found is that the messaging pop up is not announce and read by screen readers i.e. NVDA or JAWS. After testing the questioned issue, I realized after the popup was opened, the focus was gone on the close button on the top right and the button was announced, but the content was left unread. I tried on multiple browsers and the result was the same. This is while we have other popups that screen readers do announce and read them just fine.

I looked into the PayPal Javascript SDK and reviewed all the options, but couldn't find anything to enable the mentioned feature.
So I resorted to comparing HTML codes and after reviewing the working popups, I realized that they are using certain attributes to make the dialog accessible. After reviewing the PayPal iframe and applying the same changes, it started working for me. I The below code is a "Buy Now, Pay Later - PayPal" popup with only the key part that I had to modify:

``` html
<html class="">
  <head>
    <title>Buy Now, Pay Later - PayPal</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
  </head>
  <body class="modal-closed">
    ...
    <div class="modal-wrapper">
      ...
      <div class="content__wrapper-overflow " tabindex="-1" role="dialog" aria-labelledby="UNIQUE-ID-FOR-THE-CONTAINER">
        <div class="content__wrapper" id="UNIQUE-ID-FOR-THE-CONTAINER">
          ...
        </div>
      </div>
    </div>
  </body>
</html>
```
I had to add `tabindex="-1"`, `role="dialog"` and `aria-labelledby="UNIQUE-ID-FOR-THE-CONTAINER"`. Also added the `id="UNIQUE-ID-FOR-THE-CONTAINER"` to the inner HTML. There are other types of popups, but they can be fixed the same way as I tested.

I may have gone too far though and know this may not be a suitable fix, but I wanted to share my findings and have a working sample. Also, I would like to ask whether there was a specific option to enable the screen reader feature, or if not, it could be added as an ADA fix to the codebase.

Kind regards.

**Comment by Seavenly** [comment]
Hi @ali-kashanian,

Thank you for bringing this to our attention. We have a PR (#926) to address a few different modal accessibility issues and have incorporated some of these changes that should address this concern. We're working to get these changes pushed out this week. Thank you for the detailed information and steps to reproduce which were incredibly helpful for our team.

**Comment by Seavenly** [comment]
Hi @ali-kashanian,

The changes are live now so we're going to close this issue. Thanks again!
